src/util/format_expr.cpp

```diff
@@ -217,81 +217,145 @@ std::ostream &fallback_format_rec(std::ostream &os, const exprt &expr)
   return os;
 }
 
-// The below generates a string in a generic syntax
+class format_expr_configt
+{
+public:
+  format_expr_configt()
+  {
+    setup();
+  }
+
+  using formattert =
+    std::function<std::ostream &(std::ostream &, const exprt &)>;
+  using expr_mapt = std::unordered_map<irep_idt, formattert>;
+
+  expr_mapt expr_map;
+
+  /// find the formatter for a given expression
+  const formattert &find_formatter(const exprt &);
```

Comment:
Shouldn't this be `const`?

Reply:
As-is now yes, but was contemplating to generate formatters on the fly.